httpclient5/src/main/java/org/apache/hc/client5/http/impl/nio/DefaultAsyncClientConnectionOperator.java

```diff
@@ -165,7 +165,13 @@ public void completed(final IOSession session) {
                         if (tlsStrategy != null) {
                             try {
                                 final Timeout socketTimeout = connection.getSocketTimeout();
-                                final Timeout handshakeTimeout = tlsConfig.getHandshakeTimeout();
+                                // TLS handshake timeout precedence:
+                                // 1. Explicitly configured handshake timeout from TlsConfig
+                                // 2. Current socket timeout of the connection (if set)
+                                // 3. Falls back to connectTimeout if neither is specified (handled later)
+                                final Timeout handshakeTimeout = tlsConfig.getHandshakeTimeout() != null
```

Comment:
@rschmitt Would this change be OK with you? This should fix an inconsistency in the behavior of the classic and async transports.

Reply:
@rschmitt Fair enough. Please take a look at an alternative #699  and corrects the behavior of the classic connection operator instead. 

With regards to the connect timeout integration tests they consistently fail for me locally (Windows MINGW64_NT-10.0-26100). I need to get them fixed first.

Reply:
You're doing Windows development in mingw? Not WSL2 or native Windows?

Reply:
@rschmitt Ubuntu (or Debian) Linux is my primary development environment but sometimes I have to use a client issued laptop that runs Windows. I run Maven inside WinGit console. It is an old habit.